src/jsMain/kotlin/net/kyori/adventure/webui/js/Main.kt

```diff
@@ -359,6 +396,49 @@ public fun main() {
         })
 }
 
+private fun readTemplates(): Pair<Map<String, String>, Map<String, String>> {
+    val templatesBox = document.getElementById("templates-box")!!
+    val templateComponents =
+        templatesBox.getElementsByClassName("template-component").asList().map {
+            it.unsafeCast<HTMLInputElement>().checked
+        }
+    val templateKeys =
+        templatesBox.getElementsByClassName("template-key").asList().map {
+            it.unsafeCast<HTMLInputElement>().value
+        }
+    val templateValues =
+        templatesBox.getElementsByClassName("template-value").asList().map {
+            it.unsafeCast<HTMLInputElement>().value
+        }
+    val stringTemplates = mutableMapOf<String, String>()
+    val miniMessageTemplates = mutableMapOf<String, String>()
+    templateKeys
+        .zip(templateValues)
+        .zip(templateComponents)
+        .filter { (t, _) -> t.first.isNotEmpty() && t.second.isNotEmpty() }
+        .forEach { (t, c) ->
+            (if (c) miniMessageTemplates else stringTemplates)[t.first] = t.second
```

Comment:
not a comment, just a general observation that kotlin is really cursed sometimes

Reply:
![image](https://user-images.githubusercontent.com/53803019/139528266-babd8d84-7365-49cd-8596-00f8c16574af.png)